### ISSUE TITLE
Select Spool Not Work In Mainsail 

### ISSUE THREAD

**Issue by Multicore87** [Donkie/Spoolman#246]
The select spool Not work in mainsail, i try with many browser.
![IMG_3978](https://github.com/Donkie/Spoolman/assets/91397090/cb8fbad9-e1d3-4b61-a2e3-e8b02897f06b)

**Comment by pdsccode** [comment]
Did you do any updates of your mainsail instance? What does the moonraker.log show?
All in all I'd suspect that this is not a Spoolman issue but log files are helpful for clarification.

**Comment by Multicore87** [comment]
I don't know if it depends on spoolman or anything. It doesn't go with any browser while from a smartphone or clipper screen it seems so. What do you have to do? What files do you need or what do you want to know?

**Comment by Multicore87** [comment]
I also have a error on moonracker that concerns spoolman

**Comment by pdsccode** [comment]
The file of interest would be "moonraker.log" which can be found in the logs folder of your Mainsail installation
![image](https://github.com/Donkie/Spoolman/assets/42718020/8cbf5921-dba2-43cb-8eac-e24b6e4adb7e)

**Comment by Multicore87** [comment]
[moonraker.log](https://github.com/Donkie/Spoolman/files/13809497/moonraker.log)

**Comment by Multicore87** [comment]
Just put if you can understand something

**Comment by pdsccode** [comment]
> I also have a error on moonracker that concerns spoolman

Can you show me the error you have somehow? The logs seem pretty normal and there is no error seen.

**Comment by Multicore87** [comment]
[moonraker.log.2023-12-24.txt](https://github.com/Donkie/Spoolman/files/13811563/moonraker.log.2023-12-24.txt)
[moonraker.log.2024-01-01.txt](https://github.com/Donkie/Spoolman/files/13811565/moonraker.log.2024-01-01.txt)

**Comment by Donkie** [comment]
Everything seems to be working fine according to the logs, which makes me believe it's your browser. Can you check your browser's javascript console and see if theres any errors that pop up there when you open the spool selection window?

**Comment by Multicore87** [comment]
![IMG_4002](https://github.com/Donkie/Spoolman/assets/91397090/8b1412be-5f69-4679-a5b2-ff9e39c833a2)

**Comment by Multicore87** [comment]
I've tried with many browsers like chrome, Safari, Edge etc. both Mac and Windows but it doesn't go. I don't think it depends on the browser but something about spoolman or moonracker or mainsail I don't know.

**Comment by rufinus** [comment]
0.8.0-209 may be a bit to old to have the spoolman integration.

**Comment by Multicore87** [comment]
So what should I do?

**Comment by Ladin69** [comment]
``[spoolman]
server: http://192.168.0.20:7912  ( link to yours Spool Manger )
#   URL to the Spoolman instance. This parameter must be provided.
# sync_rate: 5 
#   The interval, in seconds, between sync requests with the
#   Spoolman server.  The default is 5.

**Comment by Ladin69** [comment]
```
[spoolman]
server: http://192.168.0.20:7912  ( link to yours Spool Manger )
#   URL to the Spoolman instance. This parameter must be provided.
# sync_rate: 5
#   The interval, in seconds, between sync requests with the
#   Spoolman server.  The default is 5.
```

**Comment by rufinus** [comment]
@ladin69 can you stop commenting with your config under every ticket? this neither helps the topic nur you in any way.

**Comment by Multicore87** [comment]
What can I do?

**Comment by Multicore87** [comment]
I had already tried to remove the spoolman references on mooracker and everything else but nothing to do. I can try this javascript thing if you tell me what to do but does this have to do with it since it doesn't work with any browser or operating system?

**Comment by Multicore87** [comment]
![Uploading IMG_4004.jpeg…]()

**Comment by Donkie** [comment]
It's probably nothing wrong with your browser, but obviously something is wrong with your UI, and if something is wrong with the UI then it's likely you also have some error in your browser console.

**Comment by pdsccode** [comment]
> I had already tried to remove the spoolman references on mooracker and everything else but nothing to do. I can try this javascript thing if you tell me what to do but does this have to do with it since it doesn't work with any browser or operating system?

As Donkie said, it is likely to see the issue in the javascript console even if it has nothing to do with the browser itself. https://www.digitalocean.com/community/tutorials/how-to-use-the-javascript-developer-console shows how to access the console from different browsers.

P.S. did you switch Mainsail hosts in the meantime? The first two logs show a user "pi" likely from a Raspberry Pi, the last one shows a "biqu" user from either a Biqu Hurakan or a BTT Manta board.

**Comment by Multicore87** [comment]
Because he does the same thing to both me and a friend, I have Raspberry and he cb1 btt. I'm trying to understand for both of us, and for that I think it's not just my problem but maybe it depends on mainsail or moonracker. The main problems are that if you open filament picker from blank page and every now and then it also gives error notification as in the picture

**Comment by Multicore87** [comment]
![IMG_4006](https://github.com/Donkie/Spoolman/assets/91397090/b16cd509-76d5-428d-a516-b22a6b1993a4)
![IMG_4004](https://github.com/Donkie/Spoolman/assets/91397090/a134740c-c8c4-4a5f-a41d-8ba4157d3c2f)

**Comment by pdsccode** [comment]
Alright, I see. Then please do us and yourself a favour and at least leave some info when you provide a log that is not from your instance. This can lead to confusion and makes it even harder to help.

To move forward with your issue, please check the JavaScript console as requested. Open the console, then open the "pick spool" dialog, and post the output from the console here.

**Comment by Multicore87** [comment]
hi, yes spoolman runs on a raspberry along with everything without docker. the interface with ip opens easily and everything works. I can only select or remove filaments via macro. but if I open the page to select it is completely white as in the photo.
I tried to edit in the moonracker file with ip http://127.0.0.1:7912 but the problem doesn't change.

**Comment by Multicore87** [comment]
![294262832-a134740c-c8c4-4a5f-a41d-8ba4157d3c2f](https://github.com/Donkie/Spoolman/assets/91397090/a92c8528-cf3b-4c7a-8a85-129315f52353)

**Comment by Multicore87** [comment]
this is the window select spool on all browser and operative sistems

**Comment by Multicore87** [comment]
![Javascript](https://github.com/Donkie/Spoolman/assets/91397090/eae35c0e-0668-4c5e-877b-f9286b6cea24)

**Comment by pdsccode** [comment]
Almost there. Would you please close the select spool dialog (the one that doesn't work) again? Then click on "console" in the dev tools on the right and open the spool selection again.

![image](https://github.com/Donkie/Spoolman/assets/42718020/bb272bca-2c29-46c1-9a9e-7c48fa21bae5)

Post another screenshot after you done this please.

**Comment by Multicore87** [comment]
https://github.com/Donkie/Spoolman/assets/91397090/f81ebd1e-b909-47d5-87e0-4f52c0117de1

**Comment by Multicore87** [comment]
[moonraker.log](https://github.com/Donkie/Spoolman/files/13851640/moonraker.log)

**Comment by Multicore87** [comment]
![tempImage5t0n09](https://github.com/Donkie/Spoolman/assets/91397090/451372cb-9a81-4d30-a0c2-b0b786b30160)

**Comment by Multicore87** [comment]
This error also often appears

**Comment by pdsccode** [comment]
Ok, now we have an actual error in the log file.
Just to check the current state, the Spoolman web interface is still correctly working via http://192.168.1.15:7912?

**Comment by Ladin69** [comment]
@Multicore87 You skipped something during the installation, I'm repeating the entire installation

**Comment by Multicore87** [comment]
I've already tried several times and to redo installation but the problem doesn't change

**Comment by pdsccode** [comment]
I agree with Ladin69 here to a degree. I went ahead and tried to mimic your environment as close as possible by installing a fresh MainsailOS to a Raspberry Pi and installing Spoolman standalone on the same Pi as described in the Readme.
- the Webinterface for spoolman is reachable on the same IP on port 7912 and working correctly
- After updating all components of Mainsail to the latest versions (Tab "Machine") and adding the Spoolman config entry to moonraker.conf, everything is working correctly:
 
![image](https://github.com/Donkie/Spoolman/assets/42718020/00d2594a-e8a2-4f41-a2fb-2b9a28183fc8)

Please check again if there are pending updates on the "Machine" tab in Mainsail. If not, I would suspect a broken Mainsail install at this point and suggest a complete reinstall of the Pi with a current image. Make sure to backup the printer.cfg before doing so.
